src/studio/src/designer/frontend/packages/schema-editor/src/components/Editor.tsx

```diff
@@ -174,34 +179,39 @@ export const Editor = (props: IEditorProps) => {
         </section>
         {schema ? (
           <div id='schema-editor' className={classes.editor}>
-            <TabContext value={tabIndex}>
+            <TabContext value={selectedTab}>
               <AppBar
                 position='static' color='default'
                 className={classes.appBar}
               >
                 <TabList
-                  onChange={(e, v) => setTabIndex(v)}
+                  onChange={(_x, value) => dispatch(setSelectedTab({ selectedTab: value }))}
                   aria-label='model-tabs'
                 >
                   <SchemaTab
                     label='models'
                     language={language}
                     value='0'
+                    key='0'
```

Comment:
Does setting keys here actually do anything?
I thought keys were used to uniquely identify elements in maps so they don't have to re-render.
What are the keys used for here?

Reply:
Nope, just me playing around trying to prevent expanded items from disappearing. Removed.